File: acct-currency-ws-spec/src/main/java/com/desolatetimelines/acct/currency/ws/model/MonitoredCurrencyCollector.java
package com.desolatetimelines.acct.currency.ws.model;

import java.util.Collection;

/**
 * Defines a currency collector that's available for use for the purpose of
 * gathering records for monitored currencies
 *
 * @param currencyCollectorName the name that uniquely identifies the currency exchange rates collector
 * @param supportedBankCodes    a collection of the codes of the banks that are supported by the collector
 */
public record MonitoredCurrencyCollector(
    String currencyCollectorName,
    Collection<String> supportedBankCodes
) {
}

File: acct-gui-angular/src/app/model-acct/monitored-currency-collector.ts
/**
 * Defines a currency collector that's available for use for the purpose of
 * gathering records for monitored currencies
 */
export interface MonitoredCurrencyCollector {

    /**
     * The name that uniquely identifies the currency exchange rates collector
     */
    currencyCollectorName : string

    /**
     * A collection of the codes of the banks that are supported by the collector
     */
    supportedBankCodes : string[]

}